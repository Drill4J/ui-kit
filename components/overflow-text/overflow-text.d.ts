/// <reference types="react" />
export declare const OverflowText: import("react").ComponentType<DOMAttrs>;
