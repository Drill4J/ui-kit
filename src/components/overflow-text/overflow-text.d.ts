export declare const OverflowText: import('styled-components').StyledComponent<'div', any, {}, never>;
