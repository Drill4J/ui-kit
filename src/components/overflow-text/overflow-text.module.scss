@import '../../common/style-guide/index';

.overflow-text {
  width: 100%;
  height: 1.3em;
  overflow: hidden;
  white-space: nowrap;
  position: relative;

  &::after {
    content: '';
    position: absolute;
    right: 0;
    top: 0;
    width: 20px;
    height: 100%;
    background: linear-gradient(
      to right,
      rgba(map-get($color-palette, white), 0.2),
      map-get($color-palette, white) 100%
    );
  }
}
