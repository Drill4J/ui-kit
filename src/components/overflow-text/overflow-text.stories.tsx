import React from 'react';
import { storiesOf } from '@storybook/react';

import { OverflowText } from './overflow-text';

storiesOf('Overflowext', module).add('Overflow text', () => (
  <div style={{ width: '150px' }}>
    <OverflowText>Overflow text Overflow text</OverflowText>
  </div>
));
