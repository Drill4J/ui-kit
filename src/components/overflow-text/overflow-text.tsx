import * as React from 'react';
import { BEM } from '@redneckz/react-bem-helper';

import styles from './overflow-text.module.scss';

export interface Props {
  className?: string;
  children: string;
}

const overflowText = BEM(styles);

export const OverflowText = overflowText(({ className, children }: Props) => {
  const messageRef = React.useRef<HTMLDivElement>(null);
  const [hint, setHint] = React.useState<string | null>(null);

  React.useEffect(() => {
    const observer = new IntersectionObserver(
      ([entry]) => (!entry.isIntersecting ? setHint(children) : setHint(null)),
      {
        root: null,
        threshold: 1.0,
      },
    );

    messageRef.current && observer.observe(messageRef.current);

    return () => {
      observer.disconnect();
    };
  }, []);

  return (
    <div className={className} data-title={hint}>
      <Wrapper>
        <span ref={messageRef}>{children}</span>
      </Wrapper>
    </div>
  );
});

const Wrapper = overflowText.wrapper('span');
