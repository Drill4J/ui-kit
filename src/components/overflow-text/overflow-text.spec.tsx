import React from 'react';
import renderer from 'react-test-renderer';

import { OverflowText } from './overflow-text';

it('renders correctly', () => {
  const tree = renderer.create(<OverflowText>Foo bar</OverflowText>).toJSON();
  
  expect(tree).toMatchSnapshot();
});