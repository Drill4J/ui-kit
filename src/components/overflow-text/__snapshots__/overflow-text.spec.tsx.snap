// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="overflow-text"
  data-modifiers=""
>
  Foo bar
</div>
`;

exports[`should match snapshot 1`] = `
<div
  className="overflow-text"
  data-modifiers=""
>
  Foo bar
</div>
`;
